test/e2e/config_test.go

```diff
@@ -727,6 +732,49 @@ func TestUserWorkloadMonitorThanosRulerConfig(t *testing.T) {
 	}
 }
 
+// checkMonitorConsolePluginReachable makes sure that one of the pods at least can serve /plugin-manifest.json
+// TODO: This is better done via a readiness probe in the Deployment.
+func checkMonitorConsolePluginReachable(t *testing.T, pluginName string) {
+	err := framework.Poll(time.Second, 5*time.Minute, func() error {
+		host, cleanUp, err := f.ForwardPort(t, f.Ns, pluginName, 9443)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer cleanUp()
+
+		client := &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+			},
+		}
+		resp, err := client.Get(fmt.Sprintf("https://%s/plugin-manifest.json", host))
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+
+		b, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return fmt.Errorf("fail to read response body: %w", err)
+		}
+
+		if resp.StatusCode != http.StatusOK {
+			return fmt.Errorf("expected status %d, got %d (%q)", resp.StatusCode, http.StatusOK, framework.ClampMax(b))
+		}
+
+		res, err := gabs.ParseJSON(b)
+		if err != nil {
+			return err
+		}
+		name, ok := res.Path("name").Data().(string)
+		if !ok || name != pluginName {
+			t.Errorf("expected plugin name to be %s, got `%s`", pluginName, name)
```

Comment:
(nit) if !ok then name is empty? Also shouldn't we return an err rather than calling t.Errorf()?

Reply:
Good catch for the `t.Errorf`
> (nit) if !ok then name is empty

That's why I'm putting it between ``

Reply:
`t.Errorf` still makes the test fail, but it should be cleaner now.